addons/udes_stock/models/stock_picking.py

```diff
@@ -2463,13 +2463,16 @@ def search_for_pickings(self, picking_type_id, picking_priorities, limit=1, doma
             if categories:
                 search_domain.append(("u_location_category_id", "child_of", categories.ids))
 
-        search_domain.extend(
-            [
-                ("picking_type_id", "=", picking_type_id),
-                ("state", "=", "assigned"),
-                ("batch_id", "=", False),
-            ]
-        )
+        extra_filter = [
```

Comment:
To be consistent in the way search_domain have been extended i would suggest not to use a new variable extra_filter, but directly extend search_domain if picking_type.u_batch_dest_loc_not_allowed